### PR TITLE
Check for unlikely "zero speed" condition

### LINKED ISSUE #616: Debug assertion failure: AlongStep.hh:110: `speed > 0`
#### Failure:
```json
    "failure": {
     "condition": "speed > 0",
     "file": "/home/alund/celeritas_project/celeritas/src/celeritas/global/alongstep/AlongStep.hh",
     "line": 110, 
     "type": "DebugError",
     "which": "internal assertion failed"
    }  
```
#### Regression problem:
cms2018-vecgeom-cpu
<details><summary>JSON input</summary><pre>
{
"_geometry": "vecgeom",
"_instance": 0,
"_name": [
"cms2018",
"vecgeom",
"cpu"
],
"_num_events": 7,
"_num_primaries": 9100,
"_outdir": "cms2018-vecgeom-cpu",
"_timeout": 600.0,
"brem_combined": false,
"eloss_fluctuation": true,
"enable_diagnostics": false,
"geant_options": {
"coulomb_scattering": false,
"eloss_fluctuation": false,
"em_bins_per_decade": 56, 
"lpm": true,
"msc": "none",
"physics": "em_basic",
"rayleigh_scattering": true
},
"geometry_filename": "/home/alund/celeritas_project/regression/input/cms2018.gdml",
"hepmc3_filename": "/home/alund/celeritas_project/regression/input/simple-cms-13TeV.hepmc3",
"initializer_capacity": 1048576,
"mag_field": [
0.0,
0.0,
0.0
],
"max_events": 7,
"max_num_tracks": 4096,
"max_steps": 524288,
"physics_filename": "/home/alund/celeritas_project/regression/input/cms2018.gdml",
"secondary_stack_factor": 3.0,
"seed": 20220904,
"sync": true,
"use_device": false
}
</pre></details>

#### Diagnosis:
The electron has a pre-step energy = 5.83609639577594e-03. The step is range-limited, with range = 1.00002899990615401e-01. In the range-to-step conversion, the range is slightly larger than the min range x fudge factor, so the step is scaled to step = 1.00002899923337801e-01. The mean energy loss = 5.83609639577594309e-03, resulting in a super small but nonzero post-step energy = 1.73472347597681e-18. Because the energy is so low, the calculated beta^2 is zero due to rounding error, and the speed is zero.

### PR DESCRIPTION
**Pull request by sethrj**
For very small energies the calculated speed of a non-stopped particle can be exactly zero: see the note on `celeritas::ParticleTrackView::beta_sq()`. This only occurs when the `energy / mass` ratio is less than numeric epsilon (so ~1e-17 MeV for electrons). Rather than try to improve the precision of `beta_sq` for massive particles that are nearly at rest (not relevant to the current HEP detectors we're targeting), I've just added a conditional to *not* adjust the global time when the speed is almost zero.

Fixes #616

### CODE REVIEW

**Review by amandalund — APPROVED** [review]
Looks good.

**Review by whokion — APPROVED** [review]
This is okay for now as a tweak, but please see additional comments.

File: test/celeritas/global/AlongStep.test.cc [hunk 1]
```diff
@@ -87,6 +91,52 @@ TEST_F(KnAlongStepTest, basic)
...
+        auto result = this->run(inp, num_tracks);
+        EXPECT_SOFT_EQ(1e-06, result.eloss);
+        EXPECT_SOFT_EQ(5.2704627669473e-05, result.displacement);
+        EXPECT_SOFT_EQ(1, result.angle);
+        EXPECT_SOFT_EQ(1.2431209185653e-12, result.time);
+        EXPECT_SOFT_EQ(5.2704627669473e-05, result.step);
+        EXPECT_EQ("physics-discrete-select", result.action);
+    }
+    {
+        inp.energy = MevEnergy{1e-12};
```
[the remark this answers is not in the document]
Comment (sethrj) [reply test/celeritas/global/AlongStep.test.cc:120]: They're showing up in real life because we're not applying the energy loss cut *after* energy loss. I've got an incoming PR that will fix that in the usual cases, but because it doesn't apply on the boundaries there's still a microscopic chance that we'll have a very low energy particle. (Or maybe we'll get some other roundoff issue from a post-collision.) In any case it's good to make sure the code behaves in those cases...

File: src/celeritas/global/alongstep/AlongStep.hh [hunk 2]
```diff
@@ -96,10 +96,16 @@ inline CELER_FUNCTION void along_step(MH&& msc,
     // Update track's lab-frame time using the beginning-of-step speed
     {
         auto particle = track.make_particle_view();
+        CELER_ASSERT(!particle.is_stopped());
         real_type speed = native_value_from(particle.speed());
-        CELER_ASSERT(speed > 0);
-        real_type delta_time = local.step_limit.step / speed;
-        sim.add_time(delta_time);
+        CELER_ASSERT(speed >= 0);
```
[the remark this answers is not in the document]
Comment (sethrj) [reply src/celeritas/global/alongstep/AlongStep.hh:101]: It's not quite logically consistent but I don't think it's worse than any of the other options. Assuming it's a particle with slowing-down processes, it's going to be stopped by the end of the along-step kernel, and that will naturally take care of depositing any (tiny) residual energy as well as stopping it for the discrete selection. If we stop "zero-speed" particles at the beginning of along-step, then we have to duplicate that logic but still end up incrementing zero time to the track.

Anyway, I think that skipping `time += step/0` lets the hypothetical low-energy particle stop naturally with the rest of the along-step logic. And this should be increasingly rare with #648 .
